js/coincheck.js

```diff
@@ -499,6 +501,171 @@ module.exports = class coincheck extends Exchange {
+        // }
+        const data = this.safeValue (response, 'deposits', []);
+        return this.parseTransactions (data, currency, since, limit);
+    }
+
+    async fetchWithdrawals (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+        }
+        const request = {};
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.privateGetWithdraws (this.extend (request, params));
+        //  {
+        //   "success": true,
+        //   "pagination": {
+        //     "limit": 25,
+        //     "order": "desc",
+        //     "starting_after": null,
+        //     "ending_before": null
+        //   },
+        //   "data": [
+        //     {
+        //       "id": 398,
+        //       "status": "finished",
+        //       "amount": "242742.0",
+        //       "currency": "JPY",
+        //       "created_at": "2014-12-04T15:00:00.000Z",
+        //       "bank_account_id": 243,
+        //       "fee": "400.0",
+        //       "is_fast": true
+        //     }
+        //   ]
+        // }
+        const data = this.safeValue (response, 'data', []);
+        return this.parseTransactions (data, currency, since, limit);
+    }
+
+    parseDepositStatus (status) {
+        const statuses = {
+            'confirmed': 'pending',
+            'received': 'ok',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    parseWithdrawalStatus (status) {
+        const statuses = {
+            'pending': 'pending',
+            'processing': 'pending',
+            'finished': 'ok',
+            'canceled': 'canceled',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        //
+        // fetchDeposits
+        //
+        // {
+        //       "id": 2,
+        //       "amount": "0.05",
+        //       "currency": "BTC",
+        //       "address": "13PhzoK8me3u5nHzzFD85qT9RqEWR9M4Ty",
+        //       "status": "confirmed",
+        //       "confirmed_at": "2015-06-13T08:29:18.000Z",
+        //       "created_at": "2015-06-13T08:22:18.000Z"
+        //  }
+        //
+        // fetchWithdrawals
+        //
+        //  {
+        //       "id": 398,
+        //       "status": "finished",
+        //       "amount": "242742.0",
+        //       "currency": "JPY",
+        //       "created_at": "2014-12-04T15:00:00.000Z",
+        //       "bank_account_id": 243,
+        //       "fee": "400.0",
+        //       "is_fast": true
+        //  }
+        //
+        let type = undefined;
+        let status = undefined;
+        const rawStatus = this.safeString (transaction, 'status');
+        if ('address' in transaction) {
+            type = 'deposit';
```

Comment:
Tbh, I'm not entirely sure about this logic for the `type` field – do you have an example of a crypto-withdrawal? Cause I would expect to see an `address` there as well, and then we can't use that field to determine the type of this transaction. We should probably check for the `fee` instead, or for some other indication. Also, we could drop the `type` here completely, and extend those transactions with type in `fetchDeposits` and `fetchWithdrawals` as we do here, for example:

- fetchDeposits: https://github.com/ccxt/ccxt/blob/master/js/bitpanda.js#L1044
- fetchWithdrawals: https://github.com/ccxt/ccxt/blob/master/js/bitpanda.js#L1100

↑ That will get propagated to all txs, after parsing them.

Also, I think we can merge parseDepositStatus + parseWithdrawalStatus = parseTransactionStatus like we do here:

- https://github.com/ccxt/ccxt/blob/master/js/bitget.js#L2522

I know it may be too much to ask, but if you could fix for the above – I could merge it sooner. Thanks for your help and involvement!

Reply:
No worries at all. I agree with all your comments! I have pushed a commit addressing those 😃